Meta/serenity_gdb.py

```diff
@@ -0,0 +1,263 @@
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import gdb
+import gdb.types
+import re
+
+
+class AKAtomic(object):
+    def __init__(self, val):
+        self.val = val
+
+    def to_string(self):
+        return self.val["m_value"]
+
+
+class AKDistinctNumeric(object):
+    def __init__(self, val):
+        self.val = val
+
+    def to_string(self):
+        return self.val["m_value"]
+
+
+class AKRefCounted(object):
+    def __init__(self, val):
+        self.val = val
+
+    def to_string(self):
+        return self.val["m_ref_count"]
+
+
+class AKString(object):
+    def __init__(self, val):
+        self.val = val
+
+    def to_string(self):
+        if int(self.val["m_impl"]["m_bits"]["m_value"]) == 0:
+            return "\"\""
+        else:
+            impl = AKRefPtr(self.val["m_impl"]).get_pointee().dereference()
+            return AKStringImpl(impl).to_string()
+
+
+class AKStringView(object):
+    def __init__(self, val):
+        self.val = val
+
+    def to_string(self):
+        if int(self.val["m_length"]) == 0:
+            return "\"\""
+        else:
+            characters = self.val["m_characters"]
+            str_type = characters.type.target().array(self.val["m_length"]).pointer()
+            return str(characters.cast(str_type).dereference())
+
+
+def get_field_unalloced(val, member, type):
+    # Trying to access a variable-length field seems to fail with
+    # Python Exception <class 'gdb.error'> value requires 4294967296 bytes, which is more than max-value-size
+    # This works around that issue.
+    return gdb.parse_and_eval("*(%s*)((%s*)%s)->%s" % (type, val.type.name, int(val.address), member))
```

Comment:
We can assume modern Python >= 3.6 (3.5 and below had EOL already), please use f-strings. Same below.